adapters/repos/db/vector/hnsw/startup.go

```diff
@@ -197,6 +203,44 @@ func (h *hnsw) restoreFromDisk() error {
 	return nil
 }
 
+func (h *hnsw) restoreDocMappings() error {
+	prevDocID := uint64(0)
+	relativeID := uint64(0)
+	maxNodeID := uint64(0)
+	for _, node := range h.nodes {
+		if node == nil {
+			continue
+		}
+		nodeIDBytes := make([]byte, 8)
```

Comment:
This could be used as a buffer outside the loop, and renamed to be more generic:

```go
buf := make([]byte, 8)
for _, node := range h.nodes {
  // ...
  binary.BigEndian.PutUint64(buf, node.id)
```